documentation/src/main/asciidoc/backend-lucene.asciidoc

```diff
@@ -543,3 +542,136 @@ link:http://wiki.apache.org/solr/AnalyzersTokenizersTokenFilters[Solr Wiki]
+are *not* marked for commit,
+leading to less frequent commits and thus higher write throughput.
+
+In write-intensive scenarios where a commit at the end of the batch is still too frequent,
+it is possible to commit less frequently
+and thus improve write throughput by setting a commit interval in milliseconds.
+When set to a value higher than 0, Hibernate Search will no longer commit at the end of each batch:
+if, at the end of a batch, it notices that a commit occurred less than X milliseconds ago,
+it will schedule a commit for later, when at least X milliseconds have elapsed.
+
+[WARNING]
+====
+Setting a commit interval is inherently unsafe:
+waiting 1 second before committing means that for 1 second,
+should the application crash or the server suffer a power loss,
+the changes will be lost.
+
+However, the performance boost might be worth the risk in some scenarios.
+====
+
+The commit interval is set at the index level:
+
+[source]
+----
+hibernate.search.backends.<backend name>.indexes.<index name>.io.commit_interval = 0 (default)
+# OR
+hibernate.search.backends.<backend name>.index_defaults.io.commit_interval = 0 (default)
+----
+
+[NOTE]
+====
+Remember that individual write operations may force a commit,
+which may cancel out the potential performance gains from setting a commit interval.
+
+By default, the commit interval may only improve throughput
+of the <<mapper-orm-indexing-massindexer,mass indexer>>.
+If you want changes triggered by <<mapper-orm-indexing-automatic,automatic indexing>>
+to benefit from it too, you will need to select a non-default
+<<mapper-orm-indexing-automatic-synchronization,synchronization strategy>>,
+so as not to require a commit after each change.
+====
+
+[[backend-lucene-io-refresh]]
+=== Refresh
+
+In Lucene terminology, a _refresh_ is when a new index reader is opened,
+so that the next search queries will take into account the latest changes to the index.
+
+Performance-wise, refreshing may be an expensive operation,
+which is why Hibernate Search tries not to refresh too often.
+The index reader is refreshed upon every search query,
+but only if writes have occurred since the last refresh.
+
+In write-intensive scenarios where refreshing after each write is still too frequent,
+it is possible to refresh less frequently
+and thus improve read throughput by setting a refresh interval in milliseconds.
+When set to a value higher than 0, the index reader will no longer be refreshed upon every search query:
+if, when a search query starts, the refresh occurred less than X milliseconds ago,
+then the index reader will not be refreshed, even though it may be out-of-date.
+
+The refresh interval is set at the index level:
+
+[source]
+----
+hibernate.search.backends.<backend name>.indexes.<index name>.io.refresh_interval = 0 (default)
+# OR
+hibernate.search.backends.<backend name>.index_defaults.io.refresh_interval = 0 (default)
+----
```

Comment:
I would mention the relationship between the attributes `refresh_interval ` and `synchronization.strategy`. If I understand well, with the `searchable` `synchronization.strategy` `refresh_interval` has no meaning.

Reply:
Yes, it is useful when mass indexing. But I'll cover that in [HSEARCH-3808](https://hibernate.atlassian.net/browse/HSEARCH-3808): I need to update that part of the documentation for other reasons, too.